offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/dataresources/persons/AffiliationDbConnector.java

```diff
@@ -0,0 +1,101 @@
+   * @param sessionProvider the session provider providing hibernate sessions
+   */
+  public AffiliationDbConnector(SessionProvider sessionProvider) {
+    this.sessionProvider = sessionProvider;
+  }
+
+  @Override
+  public void addAffiliation(Affiliation affiliation) throws DatabaseQueryException, AffiliationExistsException {
+    try (Session session = sessionProvider.openSession()) {
+      if (isAffiliationInSession(session, affiliation)) {
+        throw new AffiliationExistsException("The affiliation already exists.");
+      }
+      session.beginTransaction();
+      session.save(affiliation);
+    } catch (HibernateException e) {
+      log.error(e.getMessage(), e);
+      throw new DatabaseQueryException(
+          "An unexpected exception occurred during new affiliation creation");
+    }
+  }
+
+  @Override
+  public List<Affiliation> listAllAffiliations() {
+    List<Affiliation> affiliations;
+    try (Session session = sessionProvider.getCurrentSession()) {
+      session.beginTransaction();
+      Query<Affiliation> query = session.createQuery("FROM Affiliation ", Affiliation.class);
+      affiliations = new ArrayList<>(query.list());
+      session.clear();
+    }
+    return affiliations;
+  }
+
+  @Override
+  public void updateAffiliation(Affiliation affiliation)
+      throws DatabaseQueryException, AffiliationNotFoundException {
+    try (Session session = sessionProvider.openSession()) {
+      session.beginTransaction();
+      boolean affiliationWasNotFound = session.get(Affiliation.class, affiliation.getId()) == null;
+      if (affiliationWasNotFound) {
+        throw new AffiliationNotFoundException("Affiliation was not found in the database and can't be updated.");
+      }
+      session.merge(affiliation);
+      session.getTransaction().commit();
+    }
+  }
+
+  private static boolean isAffiliationInSession(Session session, Affiliation affiliation) {
+    session.beginTransaction();
+    Query<Affiliation> query =
+        session.createQuery("SELECT a FROM Affiliation a  " +
+            "WHERE a.addressAddition = :addressAddition " +
+            "AND a.category = :category " +
+            "AND a.city = :city " +
+            "AND a.country = :country " +
+            "AND a.organization = :organization " +
+            "AND a.postalCode = :postalCode " +
+            "AND a.street = :street", Affiliation.class);
+    query.setParameter("addressAddition", affiliation.getAddressAddition());
+    query.setParameter("category", affiliation.getCategory());
+    query.setParameter("city", affiliation.getCity());
+    query.setParameter("country", affiliation.getCountry());
+    query.setParameter("organization", affiliation.getOrganization());
+    query.setParameter("postalCode", affiliation.getPostalCode());
+    query.setParameter("street", affiliation.getStreet());
+
+    boolean isInSession = !query.list().isEmpty();
+    session.clear();
+    session.getTransaction().commit();
```

Comment:
I think the transaction needs to be active for query.list() to work. (But I do not know this).
The `session.clear()` is called before the commit so that no changes are committed to the database (as we only want to read and this might flush the values from the affiliation as it has the same id)